src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -416,39 +418,49 @@ public bool ForEach(KeyValuePair<string, object> activityTag)
                         PooledList<OtlpCommon.KeyValue>.Add(ref this.Tags, CreateOtlpKeyValue(key, new OtlpCommon.AnyValue { DoubleValue = d }));
                         break;
                     case int[] intArray:
+                        arrayValue = new OtlpCommon.ArrayValue();
                         foreach (var item in intArray)
                         {
-                            PooledList<OtlpCommon.KeyValue>.Add(ref this.Tags, CreateOtlpKeyValue(key, new OtlpCommon.AnyValue { IntValue = item }));
+                            arrayValue.Values.Add(new OtlpCommon.AnyValue { IntValue = item });
                         }
 
+                        PooledList<OtlpCommon.KeyValue>.Add(ref this.Tags, CreateOtlpKeyValue(key, new OtlpCommon.AnyValue { ArrayValue = arrayValue }));
                         break;
                     case double[] doubleArray:
```

Comment:
There's a number of types we're not consistently handling across everything. For example, we're not handling `float` or `float[]` in this code. Should we? Will continue discussion here https://github.com/open-telemetry/opentelemetry-dotnet/issues/2010